pkg/utils/hwaddr/hwaddr.go

```diff
@@ -20,37 +20,39 @@ import (
 )
 
 const (
-	ipRelevantByteLen      = 4
-	PrivateMACPrefixString = "0a:58"
+	ipRelevantByteLen       = 4
+	PrivateMACPrefixString  = "0a:58"
+	PrivateMACPrefixString6 = "6a:58"
 )
 
 var (
 	// private mac prefix safe to use
-	PrivateMACPrefix = []byte{0x0a, 0x58}
+	PrivateMACPrefix  = []byte{0x0a, 0x58}
+	PrivateMACPrefix6 = []byte{0x6a, 0x58}
 )
 
-type SupportIp4OnlyErr struct{ msg string }
+type InvalidIP4Err struct{ msg string }
 
```

Comment:
The invalid IPv4 error only applies to IPv4, since the GenerateHardwareAddress6() function for IPv6 operates on the current MAC address, not the IPv6 address.